File: FrontEnd/src/app/Models/profesor.model.ts
import { AsignaturaModel } from "./asignatura.model";

export class ProfesorModel {
  public idProfesor: number;
  public nombreProfesor: number;
  public apellidoProfesor: number;
  public fechaNacimientoProfesor: Date;
  public direccionProfesor: number;
  public telefonoProfesor: number;
  public identificacionProfesor: number;
  public asignaturaProfesor: AsignaturaModel[]=[];
}

File: Backend/Backend/Models/ProfesorModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Backend.Models
{
    public class ProfesorModel
    {
        public int idProfesor { get; set; }
        public int nombreProfesor { get; set; }
        public int apellidoProfesor { get; set; }
        public int fechaNacimientoProfesor { get; set; }
        public int direccionProfesor { get; set; }
        public int telefonoProfesor { get; set; }
        public int identificacionProfesor { get; set; }
        public AsignaturaModel[] asignaturaProfesor { get; set; }

    }
}